ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/MachineTestsTable.test.tsx

```diff
@@ -275,4 +288,134 @@ describe("MachineTestsTable", () => {
+      }),
+    ];
+    state.scriptresult.items = scriptResults;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <MachineTestsTable machineId="abc123" scriptResults={scriptResults} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    const checkbox = wrapper.find('input[data-test="suppress-script-results"]');
+    checkbox.simulate("change", { target: { value: "checked" } });
+
+    expect(mockSendAnalytics).toHaveBeenCalled();
+    expect(mockSendAnalytics.mock.calls[0]).toEqual([
+      "Machine testing",
+      "Unsuppress script result failure",
+      "Unsuppress",
+    ]);
+  });
+
+  it("sends an analytics event when clicking the 'View history' button", () => {
+    const scriptResults = [scriptResultFactory({ id: 1 })];
+    const scriptResultState = scriptResultStateFactory({
+      history: { 1: [partialScriptResultFactory()] },
+    });
+    state.nodescriptresult.items = { abc123: [1] };
+    state.scriptresult.items = scriptResults;
+    state.scriptresult = scriptResultState;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <MachineTestsTable machineId="abc123" scriptResults={scriptResults} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    wrapper.find("Button.p-contextual-menu__toggle").simulate("click");
+    wrapper
+      .find("Button[data-test='action-menu-show-previous']")
+      .simulate("click");
+
+    expect(mockSendAnalytics).toHaveBeenCalled();
+    expect(mockSendAnalytics.mock.calls[0]).toEqual([
+      "Machine testing",
+      "View testing script history",
+      "View previous tests",
+    ]);
+  });
+
+  it("sends an analytics event when clicking the 'View metrics' button", () => {
+    const metrics = scriptResultResultFactory({
+      title: "test-title",
+      value: "test-value",
+    });
+    const scriptResults = [scriptResultFactory({ id: 1, results: [metrics] })];
+    state.nodescriptresult.items = { abc123: [1] };
+    state.scriptresult.items = scriptResults;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <MachineTestsTable machineId="abc123" scriptResults={scriptResults} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    wrapper.find("Button.p-contextual-menu__toggle").simulate("click");
+    wrapper
+      .find("Button[data-test='action-menu-show-metrics']")
+      .simulate("click");
+
+    expect(mockSendAnalytics).toHaveBeenCalled();
+    expect(mockSendAnalytics.mock.calls[0]).toEqual([
+      "Machine testing",
+      "View testing script metrics",
```

Comment:
Should this be the same group as the history so should be "View testing script history"

Reply:
They're two separate things though?